pyshelf/routes/artifact.py

```diff
@@ -107,3 +107,33 @@ def delete_metadata_property(container, bucket_name, path, item):
         response = response_map.map_metadata_result_errors(result)
 
     return response
+
+
+@artifact.route("/<bucket_name>/artifact/_search", methods=["POST"])
+@decorators.foundation
+def root_search(container, bucket_name):
+    response = search(container)
+    return response
+
+
+@artifact.route("/<bucket_name>/artifact/<path:path>/_search", methods=["POST"])
+@decorators.foundation
+def path_search(container, bucket_name, path):
+    response = search(container)
+    return response
+
+
+def search(container):
+    """
+        Does a search with the given criteria.
+
+        Args:
+            container(pyshelf.container.Container)
+
+        Returns:
+            Flask response
+    """
+    criteria = json.loads(request.data)
```

Comment:
Better safe then sorry is what my daddy always said. 4e9ad7082ea5ce439024e6aee433f12eb270a2e7